monitor/utils/events.js

```diff
@@ -95,11 +99,39 @@ async function main(mode) {
   })).map(normalizeEvent)
 
   logger.debug("calling foreignBridge.getPastEvents('UserRequestForAffirmation')")
-  let foreignToHomeRequests = (await getPastEvents(foreignBridge, {
+  foreignToHomeRequests = (await getPastEvents(foreignBridge, {
     event: v1Bridge ? 'Withdraw' : 'UserRequestForAffirmation',
     fromBlock: MONITOR_FOREIGN_START_BLOCK,
     toBlock: foreignBlockNumber
   })).map(normalizeEvent)
+
+  // append old AMB UserRequestForSignature and UserRequestForAffirmation events
+  if (bridgeMode === BRIDGE_MODES.ARBITRARY_MESSAGE) {
+    const oldHomeBridge = new web3Home.eth.Contract(OLD_AMB_USER_REQUEST_FOR_SIGNATURE_ABI, COMMON_HOME_BRIDGE_ADDRESS)
+    const oldForeignBridge = new web3Foreign.eth.Contract(
+      OLD_AMB_USER_REQUEST_FOR_AFFIRMATION_ABI,
+      COMMON_FOREIGN_BRIDGE_ADDRESS
+    )
+
+    logger.debug("calling oldHomeBridge.getPastEvents('UserRequestForSignature(bytes)')")
+    const homeToForeignRequestsOld = (await getPastEvents(oldHomeBridge, {
```

Comment:
Could you confirm that these events will be appended to the cache the monitor workers are using if `MONITOR_CACHE_EVENTS` enabled?

Reply:
Yes, they will, the caching is done at the very end of a file, where all events are already obtained and merged together.

Reply:
Great! Thanks!